src/test/java/org/jabref/logic/exporter/CffExporterTest.java

```diff
@@ -107,30 +102,102 @@ public final void usesCorrectType(@TempDir Path tempFile) throws Exception {
+                "title: Test Title",
+                "authors:",
+                "  - family-names: Author",
+                "    given-names: Test",
+                "type: software",
+                "doi: random_doi_value");
+
+        assertEquals(expected, Files.readAllLines(file));
+    }
+
+    @Test
+    public final void exportsSoftwareDateCorrectly(@TempDir Path tempFile) throws Exception {
+        BibEntry entry = new BibEntry(StandardEntryType.Software)
+                .withField(StandardField.AUTHOR, "Test Author")
+                .withField(StandardField.TITLE, "Test Title")
+                .withField(StandardField.DATE, "2003-11-06");
+
+        Path file = tempFile.resolve("RandomFileName");
+        Files.createFile(file);
+        cffExporter.export(databaseContext, file, Collections.singletonList(entry));
+
+        List<String> expected = List.of(
+                "cff-version: 1.2.0",
+                "message: If you use this software, please cite it using the metadata from this file.",
+                "title: Test Title",
+                "authors:",
+                "  - family-names: Author",
+                "    given-names: Test",
+                "type: software",
+                "date-released: '2003-11-06'");
+
+        assertEquals(expected, Files.readAllLines(file));
+    }
+
+    @Test
+    public final void exportsArticleDateCorrectly(@TempDir Path tempFile) throws Exception {
+        BibEntry entry = new BibEntry(StandardEntryType.Article)
+                .withField(StandardField.AUTHOR, "Test Author")
+                .withField(StandardField.TITLE, "Test Title")
+                .withField(StandardField.DATE, "2003-11");
+
+        Path file = tempFile.resolve("RandomFileName");
+        Files.createFile(file);
+        cffExporter.export(databaseContext, file, Collections.singletonList(entry));
+
+        List<String> expected = List.of(
+                "cff-version: 1.2.0",
+                "message: If you use this software, please cite it using the metadata from this file.",
+                "title: Test Title",
+                "authors:",
+                "  - family-names: Author",
+                "    given-names: Test",
                 "preferred-citation:",
-                "  type: generic",
+                "  type: article",
                 "  authors:",
-                "    - name: No author specified.",
-                "  title: No title specified.");
+                "    - family-names: Author",
+                "      given-names: Test",
+                "  title: Test Title",
+                "  month: 11",
+                "  year: 2003");
 
         assertEquals(expected, Files.readAllLines(file));
     }
 
     @Test
-    void passesModifiedCharset(@TempDir Path tempFile) throws Exception {
+    public final void passesModifiedCharset(@TempDir Path tempFile) throws Exception {
```

Comment:
Why this addition of modifiers?

Side comment: `tempFile` should be named `tempDir`, because it is a directory, not a file.

Reply:
I don't understand what do you mean by addition of modifiers... This was the original test using a modified charset, I only left it as is.

Reply:
No. See the diff.